.github/workflows/ci.yml

```diff
@@ -217,11 +231,50 @@ jobs:
       - run: sudo chmod 0666 /dev/kvm
 
       - name: Test
-        run: gotestsum --raw-command --ignore-non-json-output-lines --junitfile junit.xml -- vimto -kernel :${{ matrix.tag }} -- go test -short -count 1 -json ./...
-
+        run: gotestsum --raw-command --ignore-non-json-output-lines --junitfile junit.xml -- vimto -kernel :${{ matrix.tag }} -- go test -short -count 1 -json ./... -coverprofile=coverage-${{ matrix.tag }}.out
+        
       - name: Upload Test Results
         if: always()
         uses: actions/upload-artifact@v4
         with:
           name: Test Results (${{ matrix.tag }})
           path: junit.xml
+
+      - name: Upload Coverage Report
+        if: always()
+        uses: actions/upload-artifact@v4
+        with:
+          name: Coverage Report (${{ matrix.tag }})
+          path: coverage-${{ matrix.tag }}.out
+
+  merge-coverage:
+    name: Merge Coverage Reports
+    runs-on: ubuntu-latest
+    needs: [vm-test, test-on-arm64, test-on-prev-go]
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '${{ env.go_version }}'
+
+      - name: Download Coverage Reports
+        uses: actions/download-artifact@v4
+        with:
+          pattern: coverage-*.out
+          path: coverage-reports
+
+      - name: Merge Coverage Reports
+        run: |
+          go tool covdata -merge coverage-reports/coverage-*.out -o coverage-merged.out
```

Comment:
`go tool covdata` doesn't work on the plaintext files generated with `go test -coverprofile=`. It only works with the new binary format, which is currently a bit annoying to produce with `go test`, see https://github.com/golang/go/issues/66225#issuecomment-2421289427. The cover directory needs to exist for this to work.

Documentation:
https://go.dev/doc/build-cover
https://go.dev/blog/integration-test-coverage

Please make sure your code (the commands executed, at least) work before pushing up your changes. It looks like you've been pushing commits blindly so far, which is not very respectful of a reviewer's time.

If you want to experiment with GHA workflows offline, take a look at https://github.com/nektos/act.